src/runner/highlighter.ts

```diff
@@ -17,58 +17,97 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import * as tslint from "tslint";
 import * as ts from "typescript";
+import { getComments, getText, is, toTokens } from "../utils/navigation";
 import { SonarSensor } from "./sensor";
 
 export class SyntaxHighlighter implements SonarSensor {
   public execute(sourceFile: ts.SourceFile, _: any, output: any): void {
-    new HighlighterWalker(output).walk(sourceFile);
-  }
-}
+    const highlights: HighlightedToken[] = [];
+    const tokens = toTokens(sourceFile);
 
-class HighlighterWalker extends tslint.SyntaxWalker {
-
-  private highlights: any[];
+    tokens.forEach(token => {
+      // KEYWORDS
+      if (isKeyword(token)) {
+        highlights.push(highlight(token, "keyword"));
+      }
 
-  constructor(output: any) {
-    super();
-    this.highlights = [];
-    output.highlights = this.highlights;
-  }
+      // COMMENTS
+      getComments(token).forEach(comment => {
+        highlights.push(
+          highlightComment(
+            comment,
+            getText(comment, token.getSourceFile()).startsWith("/**") ? "structured_comment" : "comment",
+            token.getSourceFile(),
+          ),
+        );
+      });
 
-  protected visitNode(node: ts.Node): void {
-    switch (node.kind) {
-      case ts.SyntaxKind.ThisKeyword : {
-        this.highlights.push(highlight(node, "k"));
-        return;
+      // STRINGS
+      const isString = is(
+        token,
+        ts.SyntaxKind.StringLiteral,
+        ts.SyntaxKind.NoSubstitutionTemplateLiteral,
+        ts.SyntaxKind.TemplateHead,
+        ts.SyntaxKind.TemplateMiddle,
+        ts.SyntaxKind.TemplateTail,
+      );
+      if (isString) {
+        highlights.push(highlight(token, "string"));
       }
-      default : {
-        if (node.getChildren().length === 0) {
-          this.highlights.push(highlight(node, "s"));
-          return;
-        } else {
-          super.visitNode(node);
-        }
+
+      // NUMBERS
+      if (is(token, ts.SyntaxKind.NumericLiteral)) {
+        highlights.push(highlight(token, "constant"));
       }
-    }
+    });
+
+    output.highlights = highlights;
```

Comment:
This is not the best practice to change the function parameter. Instead I'd `return { highlights };` and merge output from all sensors